discovery/file/file.go

```diff
@@ -23,13 +23,23 @@ func (s *FileDiscoveryService) Initialize(path string, heartbeat int) error {
 	return nil
 }
 
+func generateFromData(data []byte) []string {
+	result := make([]string, 0)
+	lines := strings.Split(strings.TrimSpace(string(data)), "\n")
+	for _, line := range lines {
```

Comment:
small nit, but could you please range over `strings.Split(strings.TrimSpace(string(data)), "\n)` directly, without creating a new variable ?